docs/components/Chips/Web.stories.tsx

```diff
@@ -121,3 +145,130 @@ Selection.parameters = {
     },
   },
 };
+
+const ControlledSelectionTemplate: ComponentStory<typeof Chips> = () => {
+  const [selected, setSelected] = useState<string[]>([
+    "Luke Skywalker",
+    "Leia Organa",
+  ]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [displayedOptions, setDisplayedOptions] = useState([
+    "Luke Skywalker",
+    "Leia Organa",
+    "Han Solo",
+    "Chewbacca",
+    "R2-D2",
+    "C-3PO",
+    "Darth Vader",
+  ]);
+
+  // Full list of all available options
+  const allOptions = [
+    "Luke Skywalker",
+    "Leia Organa",
+    "Han Solo",
+    "Chewbacca",
+    "R2-D2",
+    "C-3PO",
+    "Darth Vader",
+    "Obi-Wan Kenobi",
+    "Yoda",
+    "Emperor Palpatine",
+    "Boba Fett",
+    "Lando Calrissian",
+    "Jabba the Hutt",
+    "Admiral Ackbar",
+    "Wedge Antilles",
+  ];
+
+  const handleSelect = (value: string[]) => {
+    setSelected(value);
+  };
+
+  // Handle search in controlled mode
+  // Here we manually filter our options based on search term
+  const handleSearch = (value: string) => {
+    setSearchTerm(value);
+
+    if (value === "") {
+      setDisplayedOptions(allOptions.slice(0, 7)); // Reset to initial set
+    } else {
+      // Filter options based on search term
+      // Note: We don't need to include selected options in the filtered list
+      // The component will ensure selected chips remain visible
+      const filteredOptions = allOptions.filter(option =>
+        option.toLowerCase().includes(value.toLowerCase()),
+      );
```

Comment:
here is where one would plop in all their logic

you've received the search term, now it's up to you to see if that term exists in 1+ fields - ideally in a performant way such that you're not filtering N x M amount of times where N is the number of fields you have to search for a given entry and M is the amount of entries